backend/backend.go

```diff
@@ -27,6 +29,22 @@ type Backend interface {
 	State() (state.State, error)
 }
 
+// MultiState is an interface that a backend can implement to allow changing
+// between named states depending on the configured environment.
+type MultiState interface {
+	// States returns a list of configured named states and the current state.
+	States() ([]string, string, error)
```

Comment:
Ok, this makes sense, and I clears up part of what I wanted to cover in this first review. The original spec called for a remote ChangeState method, which implied the backend also managed the current state. Deferring to the backend was much less intrusive in the terraform code, not even accounting for possible synchronization issues between the local environment and backend.  Removing the backend from being aware of "current" at all makes this work much better.